query/physicalplan/aggregate.go

```diff
@@ -185,6 +185,37 @@ func hashCombine(lhs, rhs uint64) uint64 {
 	return lhs ^ (rhs + 0x9e3779b9 + (lhs << 6) + (lhs >> 2))
 }
 
+// hashCombiner combines a given hash with another hash that is passed.
+type hashCombiner interface {
+	hashCombine(rhs uint64) uint64
```

Comment:
hash combining is something that we do _a lot_ in the hot path, could we run a quick benchmark what this added virtualization costs us? (we can probably just add an explicit devirtualization if this has a perf impact)